src/mapping/impl/RBFParameterTuner.hpp

```diff
@@ -0,0 +1,228 @@
+   *  - void Solver::rebuildKernelDecomposition(const IndexContainer &inputIds, double radius);
+   *  - ErrorEstimate Solver::computeErrorEstimate(const Eigen::VectorXd &inputData, const IndexContainer &inputIds);
+   * @param[in] inputIds Index container of the vertices on the input mesh.
+   * @param[in] inputData Input values for which the optimizer tries to reduce the mapping error by invoking the computeErrorEstimate() method.
+   */
+  template <typename IndexContainer, typename Solver>
+  Sample optimize(Solver &solver, const mesh::PtrMesh inMesh, const IndexContainer &inputIds, const Eigen::VectorXd &inputData);
+
+  /// Returns true if the last sample tested by the optimizer also corresponds to the found optimum.
+  bool lastSampleWasOptimum() const;
+
+  double getLastOptimizedRadius() const;
+  double getLastOptimizationError() const;
+
+private:
+  /// Check if the bisection algorithm should continue based on @ref POS_TOLERANCE, @ref ERR_TOLERANCE and @ref MAX_BISEC_ITERATIONS
+  static bool shouldContinue(const Sample &lowerBound, const Sample &upperBound);
+};
+
+inline RBFParameterTuner::RBFParameterTuner(mesh::Mesh &inputMesh)
+{
+  _lowerBound = estimateMeshResolution(inputMesh);
+  _upperBound = std::numeric_limits<double>::quiet_NaN();
+
+  _lastSampleWasOptimum = false;
+}
+
+template <typename IndexContainer, typename Solver>
+Sample RBFParameterTuner::optimize(Solver &solver, const mesh::PtrMesh inMesh, const IndexContainer &inputIds, const Eigen::VectorXd &inputData)
+{
+  constexpr bool radiusRBF = Solver::BASIS_FUNCTION_T::hasCompactSupport();
+  static_assert(radiusRBF, "RBF is not supported by this optimizer, as it does not accept a support-radius."
+                           "Currently supported: Compactly supported RBFs and Gaussians.");
+
+  // To determine the _upperBound we exponentially increase the support radius by increaseSize starting with _lowerBound
+  // and check if the condition number is too high or the decomposition fails.
+  // TODO: maybe make dependent on domain extent.
+  constexpr double increaseSize = 10;
+  double           sampleRadius = _lowerBound;
+
+  _lastSampleWasOptimum = false;
+
+  PRECICE_INFO("Starting optimization with lower bound = {:.4e}, upper bound = {:.4e}", _lowerBound, _upperBound);
+
+  solver.rebuildKernelDecomposition(inMesh, inputIds, sampleRadius);
+  RBFErrorEstimate estimate   = solver.computeErrorEstimate(inputData, inputIds);
+  Sample           lowerBound = {sampleRadius, estimate.error};
+
+  // error is numerically insignificant: Either zero data or fully explained by the polynomial
+  if (estimate.error < 1e-15) {
+    _lastSampleWasOptimum = true;
+    _currentOptimum       = lowerBound;
+    return lowerBound;
+  }
+
+  // collect samples with exponential growth and decrease growth rate until the support radius is "good enough"
+  while (std::isnan(_upperBound)) {
```

Comment:
`std::isnan` depends on compiler settings. We should not introduce more reliance on this (and inf) if we ever want to support fast math modes.

Can you use some other way to signal that this hasn't been computed yet? Either a `bool` or `std::optional<double>`?